pkg/ingress/admission/api/types.go

```diff
@@ -0,0 +1,19 @@
+package api
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// IngressAdmissionConfig is the configuration for the the ingress
+// controller limiter plugin. It changes the behavior of ingress
+//objects to behave better with openshift routes and routers.
+//*NOTE* Disabling this plugin causes ingress objects to behave
+//the same as in upstream kubernetes
+type IngressAdmissionConfig struct {
+	unversioned.TypeMeta
+
+	//UpstreamHostnameUpdate when true causes updates that attempt
+	//to add or modify hostnames to succeed. Otherwise those updates
+	//fail in order to ensure hostname behavior
```

Comment:
booleans declared but not initialized are given false in golang. The code is structured so that if other options are added to IngressAdmissionConfig without enabling hostname updates they fail